### PR TITLE
Include Date methods on Common

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#41871](https://central.tri.be/issues/41871)

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task &quot;move timezones to Common&quot; | [date.php:183](https://github.com/moderntribe/tribe-common/blob/9163d39e5893493fdf2ac382e505f075930b758e/src/functions/template-tags/date.php#L183)
:warning: | Comment refers to a TODO task &quot;move timezones to Common&quot; | [date.php:227](https://github.com/moderntribe/tribe-common/blob/9163d39e5893493fdf2ac382e505f075930b758e/src/functions/template-tags/date.php#L227)
:warning: | Comment refers to a TODO task &quot;move timezones to Common&quot; | [date.php:271](https://github.com/moderntribe/tribe-common/blob/9163d39e5893493fdf2ac382e505f075930b758e/src/functions/template-tags/date.php#L271)
:warning: | Comment refers to a TODO task &quot;move timezones to Common&quot; | [date.php:312](https://github.com/moderntribe/tribe-common/blob/9163d39e5893493fdf2ac382e505f075930b758e/src/functions/template-tags/date.php#L312)

> via jenkins/codesniffer-phpcs

**Comment by borkweb** [comment]
These functions aren't useful outside of `the-events-calendar` and its plugins, right? `_EventAllDay` comes from that plugin. Because of _that_, should these really be in `tribe-common` since they aren't generic?

**Comment by bordoni** [comment]
Kinda, because we already fetch meta when we are dealing with tickets and the parent post Dates.

File: src/functions/template-tags/date.php [hunk 1]
```diff
@@ -148,3 +148,172 @@ function tribe_get_datetime_separator( $default = ' @ ', $esc = false ) {
...
+	 * Returns the event start time
+	 *
+	 * @category Events
+	 * @param int    $event       (optional)
+	 * @param string $dateFormat  Allows date and time formating using standard php syntax (http://php.net/manual/en/function.date.php)
+	 * @param string $timezone    Timezone in which to present the date/time (or default behaviour if not set)
+	 *
+	 * @return string|null Time
+	 */
+	function tribe_get_start_time( $event = null, $dateFormat = '', $timezone = null ) {
```
Comment (barryhughes) [inline src/functions/template-tags/date.php]: I realize this is existing code and you're just relocating it ... but here and elsewhere we could use this opportunity to switch to snake_cased args (totally optional).

File: src/functions/template-tags/date.php [hunk 2]
```diff
@@ -148,3 +148,172 @@ function tribe_get_datetime_separator( $default = ' @ ', $esc = false ) {
...
+
+		if ( is_numeric( $event ) ) {
+			$event = get_post( $event );
+		}
+
+		if ( ! is_object( $event ) ) {
+			return;
+		}
+
+		if ( Tribe__Date_Utils::is_all_day( tribe_get_event_meta( $event->ID, '_EventAllDay', true ) ) ) {
```
Comment (barryhughes) [inline src/functions/template-tags/date.php]: Hmm, coming back to Matt's point, isn't this line dependent on a function that lives in TEC (ie, `tribe_get_event_meta()`)? Ditto in the other funcs.

Comment (barryhughes) [inline src/functions/template-tags/date.php]: At the minimum we'd need to 'guard' this like we're doing with calls to `Tribe__Events__Timezones` methods, still, gives me pause for thought on moving this into commons.

**Pull request merged**